distribution/docker/docker-build-context/build.gradle

```diff
@@ -0,0 +1,3 @@
+apply plugin: 'base'
+
+assemble.dependsOn ":distribution:docker:buildDockerBuildContext"
```

Comment:
Why are we adding new projects for thees ?
I'm not sure what value these bring.

Reply:
@atorok It's the same structure as all of our other projects in distribution that produce artifacts, the only difference being there are explicit Gradle build files here instead of being wired up through Gradle magic in a root Gradle build file.

Reply:
thanks for the explanation!

Reply:
The reason we have individual projects is so each of those projects has the one artifact as the default artifact. Here we aren't actually consuming this build context anywhere in tests that need to depend on one or the other right? In fact, we can likely get rid of the layers we have distribution now that we don't use dependency substitution anymore for them.

The thing that gives me pause most is having a hard dependency  back to the `:distribution:docker` project on 2 tasks there. I don't see any benefit to running `:distribution:docker:docker-build-context:assemble` instead of `:distribution:docker:buildDockerBuildContext`

Reply:
@rjernst I pushed a commit that I think addresses your concerns. The back dependency is gone.